test/integration/features/profile.feature

```diff
@@ -2,104 +2,229 @@
+  Scenario: As a user, I can start profile 'foo'
+    Given profile "foo" has state "Does Not Exist"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift start" succeeds
       Then stdout should contain
       """
-      Profile 'minishift' set as active profile.
+      Add-on 'registry-route' created docker-registry route.
       """
-Scenario: User should not be able to create profile with blank profile name
-    When executing "minishift profile set"
-     Then exitcode should equal "1"
-     Then stderr should contain
+      And profile "foo" should have state "Running"
+  
+  @minishift-only
+  Scenario: As user, I can list add-ons in profile 'foo' 
+    Given profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift addons list"
+     Then stdout should match "admin-user\s*: disabled\s*P\(0\)"
+      And stdout should match "anyuid\s*: disabled\s*P\(0\)"
+      And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+      And stdout should match "xpaas\s*: disabled\s*P\(0\)"
+
+  Scenario: Checking that memory value was applied on profile 'foo'
+    When executing "minishift ssh -- less /proc/meminfo" succeeds
+     Then stdout should match "MemTotal:\s*4[0-1][0-9]{5}\s*kB"
+
+  Scenario: Checking that disk-size value was applied on profile 'foo'
+    When executing "minishift ssh -- sudo fdisk -l | grep Disk" succeeds
+     Then stdout should match "Disk \/dev\/sda: 3[0-2]\.?[0-9]{0,2} (GB|GiB)"
+
+  Scenario: Checking that cpus value was applied on profile 'foo'
+    When executing "minishift ssh -- cat /proc/cpuinfo" succeeds
+     Then stdout should match "processor\s*: 0"
+      And stdout should match "processor\s*: 1"
+      And stdout should match "processor\s*: 2"
+      And stdout should match "processor\s*: 3"
+      And stdout should not match "processor\s*: [4-9]"
+
+  Scenario: As user, I can execute command on non active running profile 'minishift' from active profile 'foo'
+    Given profile "foo" has state "Running"
+     And executing "minishift profile set foo" succeeds
+     And stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+     When executing "minishift --profile minishift stop" succeeds
+     Then stdout should contain
+     """
+     Stopping local OpenShift cluster...
+     Cluster stopped.
      """
-     A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+     And executing "minishift --profile minishift status" succeeds
+      Then stdout should match "Minishift:\s*Stopped\nProfile:\s*minishift\nOpenShift:\s*Stopped"
+
+  Scenario Outline: As user, I can delete all created profiles
+    When executing "minishift profile delete <profilename> --force" succeeds
```

Comment:
Here it is quite hidden that you are deleting also the VM with the deletion of the profile. This really needs to go to a separate scenario. It is big difference when one is deleting profiles with not existing VM and profiles with actually running VM.

Reply:
Nothing is hidden here. AFAIU whatever the profile is created of status ```Does Not Exist|Running|Stopped``` it really does not mater what is happening while deleting as we are not capturing any process in the background. The only validation ```Profile '<profilename>' deleted successfully``` is done in ```stdout``` which would be same for all the profile.
```
dhcp-0-120:v1.9.0 amit$ ./minishift profile list
- asdf		Running
- minishift	Running
- wer		Does Not Exist	(Active)
```
```
dhcp-0-120:v1.9.0 amit$ ./minishift profile delete wer --force
VM for profile 'wer' does not exist
Deleted:  /Users/amit/.minishift/profiles/wer
Profile 'wer' deleted successfully
Switching to default profile 'minishift' as the active profile.
```
```
dhcp-0-120:v1.9.0 amit$ ./minishift profile delete asdf --force
Deleted:  /Users/amit/.minishift/profiles/asdf
Profile 'asdf' deleted successfully
```

Reply:
@amitkrout I think these are two different cases:

- deleting profile which has not any VM Running or Stopped
- deleting profile which has VM Running or Stopped

In second case the Minishift has somehow first delete the VM, then has to delete the profile. Would be nice to keep these scenarios separately I think, wdyt?

Reply:
However it is true that the validation will be the same. But the 2 scenarios will describe the situation more clearly I think and in case of any failure on one of them, it would be clear that their difference somehow matters.

Reply:
I spend some time to know what really is happening. I took some sample of Active/running/does not exist profiles then i ran various scenarios on profile delete subcommand. 
```
Case 1: 
Deleting profile having no VM created from a active profile
$ ./minishift profile delete pro --force 
Profile 'pro' deleted successfully.
```
```
Case 2: 
Deleting profile having no VM created from a active profile having VM up and running
$ ./minishift profile delete pro1 --force 
Profile 'pro' deleted successfully.
```
```
Case 3: 
Deleting the active profile having no VM
$ ./minishift profile delete pro3 --force 
Profile 'pro1' deleted successfully.
Switching to default profile 'minishift' as the active profile.
```
```
Case 3: 
Deleting the profile from default active profile minishift (Both the profile has VM up and running)
$ ./minishift profile delete pro4 --force 
Profile ‘pro4’ deleted successfully.
```
In all the above cases ```Profile <profilename> deleted successfully``` is common printed output and no other output except ```Switching to default profile 'minishift' as the active profile.``` like case 2

So no need to split the scenario. Then by adding a precondition to the very next step such as ```Given profile ”minishift” is an active``` will resolve the concern.

NOTE : I executed the cases on master